R/object_name_linters.R

```diff
@@ -275,11 +273,12 @@ loweralnum <- rex(one_of(lower, digit))
 upperalnum <- rex(one_of(upper, digit))
 
 style_regexes <- list(
-  "CamelCase" = rex(start, maybe("."), upper, zero_or_more(alnum), end),
-  "camelCase" = rex(start, maybe("."), lower, zero_or_more(alnum), end),
-  "snake_case"     = rex(start, maybe("."), some_of(lower, digit), any_of("_", lower, digit), end),
-  "SNAKE_CASE"     = rex(start, maybe("."), some_of(upper, digit), any_of("_", upper, digit), end),
-  "dotted.case"    = rex(start, maybe("."), one_or_more(loweralnum), zero_or_more(dot, one_or_more(loweralnum)), end),
+  "symbols"     = rex(start, maybe("."), zero_or_more(none_of(alnum)), end),
```

Comment:
This allows a single dot and empty names. `rex(start, one_or_more(none_of(alnum)), end)` sounds like a cleaner approach.

Reply:
i thought of that but empty names won't parse:

    `` = 2
    # Error: attempt to use zero-length variable name
    assign("", 2) # same

but point taken that it's cleaner